### DISCUSSION TITLE
Idea: Establishing the proposal process

### DISCUSSION THREAD

**Discussion by galligan** [cxndao/constellation#14] [Governance]
## Overview

As the Governance committee, we are tasked with introducing a process for proposals to the DAO.

## Proposed Solution

I propose that we should heavily base our Constellation DAO Improvement Proposal process (CIPs) around the Ethereum Improvement Proposal (EIPs), beginning with a fork of EIP-1, which describes the EIP process.

The core of the EIP process is its workflow which is described as follows:

> **Idea** - An idea that is pre-draft. This is not tracked within the EIP Repository.
> 
> **Draft** - The first formally tracked stage of an EIP in development. An EIP is merged by an EIP Editor into the EIP repository when properly formatted.
> 
> **Review** - An EIP Author marks an EIP as ready for and requesting Peer Review.
> 
> **Last Call** - This is the final review window for an EIP before moving to `Final`. An EIP editor will assign `Last Call` status and set a review end date (`last-call-deadline`), typically 14 days later.
> 
> If this period results in necessary normative changes it will revert the EIP to `Review`.
> 
> **Final** - This EIP represents the final standard. A Final EIP exists in a state of finality and should only be updated to correct errata and add non-normative clarifications.
> 
> **Stagnant** - Any EIP in `Draft` or `Review` or `Last Call` if inactive for a period of 6 months or greater is moved to `Stagnant`. An EIP may be resurrected from this state by Authors or EIP Editors through moving it back to `Draft` or it's earlier status. If not resurrected, a proposal may stay forever in this status. 
> 
> >*EIP Authors are notified of any algorithmic change to the status of their EIP*
> 
> **Withdrawn** - The EIP Author(s) have withdrawn the proposed EIP. This state has finality and can no longer be resurrected using this EIP number. If the idea is pursued at later date it is considered a new proposal.
> 
> **Living** - A special status for EIPs that are designed to be continually updated and not reach a state of finality. This includes most notably EIP-1.

In our case CIP ideas could be discussed here in GitHub discussions to enable a more nuanced discussion. From there they can be proposed as drafts via a pull request. Members who are unfamiliar with GitHub should be assisted by others. Once a draft is considered final, and given a number by the Governance committee (or a CIP editorial subcommittee) then it can be submitted to Snapshot to vote.

## Next steps

If the above sounds reasonable, then I will draft our own version of EIP-1 that's best fit for our purposes and we can move to vote on it.

**Comment by mattjoyal** [comment]
> Ethereum Improvement Proposal (EIPs)

Thanks for putting this together. So we are talking about this as the template? I'm good with going with something that has been done & improved upon in the past by others so no objection here. https://github.com/ethereum/EIPs/blob/master/eip-template.md

And fork begins from here? https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1.md

Making sure I get this right, all proposal related conversation will happen in GitHub via Discussion section & then in a CIP repository? Proposal process does not encounter Discord at any point?

**Comment by galligan** [comment]
@mattjoyal 

> And fork begins from here? https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1.md

Yep. I've started in on that with #17 

> all proposal related conversation will happen in GitHub via Discussion section & then in a CIP repository? Proposal process does not encounter Discord at any point?

I dig into this a little bit with CIP-1 but those kinds of discussions can happen wherever. GitHub Discussions and Discord would be encouraged. Once the CIP idea hits the draft stage though, all of the discussion migrates to the GitHub Pull Request since it's been finalized.

**Reply by mattjoyal** [reply]
Great, makes sense, thanks for that.
